File: play-services-core/src/main/java/com/google/android/gms/constellation/PhoneNumberInfo.java
/*
 * Copyright (C) 2013-2026 microG Project Team
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.android.gms.constellation;

import android.os.Bundle;
import android.os.Parcel;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.gms.common.internal.safeparcel.AbstractSafeParcelable;
import com.google.android.gms.common.internal.safeparcel.SafeParcelable;
import com.google.android.gms.common.internal.safeparcel.SafeParcelableCreatorAndWriter;

@SafeParcelable.Class
public class PhoneNumberInfo extends AbstractSafeParcelable {
    public static SafeParcelableCreatorAndWriter<PhoneNumberInfo> CREATOR = findCreator(PhoneNumberInfo.class);

    @SafeParcelable.Field(1)
    public final int version;

    @SafeParcelable.Field(2)
    @Nullable
    public final String phoneNumber;

    @SafeParcelable.Field(3)
    public final long verificationTime;

    @SafeParcelable.Field(4)
    @Nullable
    public final Bundle extras;

    @SafeParcelable.Constructor
    public PhoneNumberInfo(
            @SafeParcelable.Param(1) int version,
            @SafeParcelable.Param(2) @Nullable String phoneNumber,
            @SafeParcelable.Param(3) long verificationTime,
            @SafeParcelable.Param(4) @Nullable Bundle extras) {
        this.version = version;
        this.phoneNumber = phoneNumber;
        this.verificationTime = verificationTime;
        this.extras = extras;
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        CREATOR.writeToParcel(this, dest, flags);
    }
}

File: play-services-constellation/src/main/kotlin/com/google/android/gms/constellation/PhoneNumberInfo.kt
package com.google.android.gms.constellation

import android.os.Bundle
import android.os.Parcel
import com.google.android.gms.common.internal.safeparcel.AbstractSafeParcelable
import com.google.android.gms.common.internal.safeparcel.SafeParcelable
import com.google.android.gms.common.internal.safeparcel.SafeParcelable.Constructor
import com.google.android.gms.common.internal.safeparcel.SafeParcelable.Field
import com.google.android.gms.common.internal.safeparcel.SafeParcelable.Param
import com.google.android.gms.common.internal.safeparcel.SafeParcelableCreatorAndWriter

@SafeParcelable.Class
data class PhoneNumberInfo @Constructor constructor(
    @JvmField @Param(1) @Field(1) val version: Int,
    @JvmField @Param(2) @Field(2) val phoneNumber: String?,
    @JvmField @Param(3) @Field(3) val verificationTime: Long,
    @JvmField @Param(4) @Field(4) val extras: Bundle?
) : AbstractSafeParcelable() {

    override fun writeToParcel(out: Parcel, flags: Int) {
        CREATOR.writeToParcel(this, out, flags)
    }

    companion object {
        @JvmField
        val CREATOR = findCreator(PhoneNumberInfo::class.java)
    }
}
